src/ott/solvers/linear/sinkhorn.py

```diff
@@ -42,6 +44,21 @@
 
 __all__ = ["Sinkhorn", "SinkhornOutput", "solve"]
 
+ProgressCallbackFn = Callable[
+    [Tuple[np.ndarray, np.ndarray, np.ndarray, NamedTuple]], None]
+"""
+Signature of any user-provided function which side effect is to report progress (print, etc).
+Such a function is called during solver iterations via
+``jax.host_callback.id_tap`` so the solver execution remains jittable.
+The arguments are each wrapped in a jax Array in order to avoid wasting
+conversion time on jax side:
+  - the current iteration: an integer wrapped in a jax Array
```

Comment:
I would only mention the wrapping only once instead for each argument, or not mention it at all (very technical and can be seen from the type hint).
Also are you sure about it always being a jax array? Locally (no jit/vmap transform), it really depends how the arguments are passed and what transformations are applied, i.e., Python ints are still ints if passed in a named tuple, however passed raw, they are converted to numpy array; jax arrays are retained.